shared/teams/container.tsx

```diff
@@ -74,56 +66,48 @@ class Reloadable extends React.PureComponent<Props & {loadTeams: () => void; onC
-        dispatch(
-          FsConstants.makeActionForOpenPathInFilesTab(FsTypes.stringToPath(`/keybase/team/${teamname}`))
-        ),
-      onReadMore: () => {
-        openURL('https://keybase.io/blog/introducing-keybase-teams')
-      },
-      onViewTeam: (teamname: Types.Teamname) =>
-        dispatch(ownProps.safeNavigateAppendPayload({path: [{props: {teamname}, selected: 'team'}]})),
-    }),
-
-    (stateProps, dispatchProps, _: OwnProps) => ({
-      deletedTeams: stateProps._deletedTeams,
-      loaded: stateProps.loaded,
-      newTeams: [...stateProps._newTeams],
-      sawChatBanner: stateProps.sawChatBanner,
-      teamNameToCanManageChat: stateProps._teamNameToRole.map(role => role !== 'none').toObject(),
-      teamNameToIsOpen: stateProps._teamNameToIsOpen.toObject(),
-      teamToRequest: makeTeamToRequest(stateProps._newTeamRequests),
-      teammembercounts: stateProps._teammembercounts.toObject(),
-      teamnames: stateProps.teamnames,
-      teamresetusers: stateProps._teamresetusers.toObject(),
-      ...dispatchProps,
-    })
-  )(Reloadable)
-)
+const _Connected = Container.connect(
+  (state: Container.TypedState) => ({
+    _deletedTeams: state.teams.deletedTeams,
+    _newTeams: state.teams.newTeams,
+    _teamresetusers: state.teams.teamNameToResetUsers || I.Map(),
+    _teams: state.teams.teamDetails,
+    loaded: !WaitingConstants.anyWaiting(state, Constants.teamsLoadedWaitingKey),
+    newTeamRequests: state.teams.newTeamRequests,
+    sawChatBanner: state.teams.sawChatBanner || false,
+  }),
+  (dispatch: Container.TypedDispatch, ownProps: OwnProps) => ({
+    ...headerActions(dispatch, ownProps),
+    loadTeams: () => dispatch(TeamsGen.createGetTeams()),
+    onClearBadges: () => dispatch(TeamsGen.createClearNavBadges()),
+    onHideChatBanner: () =>
+      dispatch(GregorGen.createUpdateCategory({body: 'true', category: 'sawChatBanner'})),
+    onManageChat: (teamname: Types.Teamname) =>
+      dispatch(
+        ownProps.safeNavigateAppendPayload({path: [{props: {teamname}, selected: 'chatManageChannels'}]})
+      ),
+    onOpenFolder: (teamname: Types.Teamname) =>
+      dispatch(
+        FsConstants.makeActionForOpenPathInFilesTab(FsTypes.stringToPath(`/keybase/team/${teamname}`))
+      ),
+    onReadMore: () => {
+      openURL('https://keybase.io/blog/introducing-keybase-teams')
+    },
+    onViewTeam: (teamname: Types.Teamname) =>
+      dispatch(ownProps.safeNavigateAppendPayload({path: [{props: {teamname}, selected: 'team'}]})),
+  }),
+  (stateProps, dispatchProps, _: OwnProps) => ({
+    deletedTeams: stateProps._deletedTeams,
+    loaded: stateProps.loaded,
+    newTeams: stateProps._newTeams,
+    newTeamRequests: stateProps.newTeamRequests,
+    sawChatBanner: stateProps.sawChatBanner,
+    teamresetusers: stateProps._teamresetusers.toObject(),
+    teams: orderTeams(stateProps._teams),
+    ...dispatchProps,
+  })
+)(Reloadable)
+const Connected = Container.withSafeNavigation(_Connected)
```

Comment:
Separated these to help debugging.

Reply:
also note withSafeNavigation is currently not typed